cameraview/src/main/java/com/otaliastudios/cameraview/engine/CameraEngine.java

```diff
@@ -713,6 +714,8 @@ public abstract void startAutoFocus(@Nullable Gesture gesture,
 
     public abstract boolean isTakingVideo();
     public abstract void takeVideo(@NonNull VideoResult.Stub stub, @NonNull File file);
+    public abstract void takeVideo(@NonNull VideoResult.Stub stub,
+                                   @NonNull FileDescriptor fileDescriptor);
```

Comment:
Instead of adding a new `takeVideo` here (which in the end forces you to have 3 of them), can we change the existing signature to accept both file and file descriptor? 
This will simplify logic in both CameraBaseEngine and CameraView.